src/compiler/checker.ts

```diff
@@ -2297,7 +2308,17 @@ namespace ts {
                             }
                         }
                         else {
-                            error(name, Diagnostics.Module_0_has_no_exported_member_1, moduleName, declarationName);
+                            if (moduleSymbol.exports && moduleSymbol.exports.has("default" as __String)) {

```

Comment:
Judging by other code in the checker, I think this could be more idiomatic.

- can you try `.has(InternalSymbolName.Default)` instead of a hard-coded string? Then it's more consistent with the surrounding code and we don't need a cast.